Document unreleased MTP telemetry items from main (#54342)
dotnet/docs: docs/core/testing/microsoft-testing-platform-telemetry.md

| All | Version of your test adapter. |
| All | Guid to correlate events from a single runner. |
| 1.0.3 | Guid to correlate events from a single test run. |
| (main) | Timestamp when test adapter loading started. |
| (main) | Timestamp when test adapter loading completed. |
| (main) | Usage statistics for MSTest attributes, as JSON. |
| (main) | Timestamp when test host builder execution started. |
| (main) | Timestamp when test host builder execution completed. |
| (main) | Count of assertion methods used, as JSON. |
| (main) | Timeout in milliseconds for assembly cleanup. |
| (main) | Timeout in milliseconds for assembly initialization. |
| (main) | Boolean for capturing debug traces. |
| (main) | Timeout in milliseconds for class cleanup. |
| (main) | Timeout in milliseconds for class initialization. |
| (main) | Boolean for treating empty data sources as inconclusive. |

> | (main) | Boolean for enabling cooperative cancellation. |

Reviewer: Minor — `mstest.setting.cooperative_cancellation` is `AsTelemetryBool(settings.CooperativeCancellationTimeout)`. It reports whether a cooperative-cancellation timeout was configured, not whether the feature is "enabled" per se. Consider tweaking to e.g. "Boolean indicating if a cooperative cancellation timeout was configured." Not blocking.

> | (main) | Boolean for enabling cooperative cancellation. |

Author: `main` is on purpose, this is ready for next release. Then it should be replaced with the actual version.

> | (main) | Source of MSTest configuration (runsettings or testconfig). |

Reviewer: The source code comment for `mstest.config_source` lists three possible values, not two: *"runsettings, testconfig.json, or none"*. Suggest tightening to enumerate all three and wrap literals in inline code to match prevailing table style:

```suggestion
| (main) | Source of MSTest configuration (`runsettings`, `testconfig.json`, or `none`). |
```

| (main) | Timestamp when test host creation started. |
| (main) | Timestamp when test host creation completed. |

> | (main) | Count of custom test method types used. |

Reviewer: This isn''t a count — `mstest.custom_test_method_types` is built by `SerializeCollection(customMethodTypesSnapshot)` and emitted as a JSON array of anonymized type-name hashes (see `AddDiscoveryMetrics` in `MSTestTelemetryDataCollector.cs`). Suggest matching the wording style of the existing "Usage statistics for MSTest attributes, as JSON." row:

```suggestion
| (main) | Anonymized list of custom test method types used, as JSON. |
```

Also: the same source method emits `mstest.custom_test_class_types` (mirror of the method-types collection), but it''s not documented in this PR — worth adding alongside this row.

| (main) | Boolean indicating if application exited gracefully. |
| (main) | Timestamp when test execution load phase started. |

> | (main) | Timestamp when test execution load phase completed. |

Reviewer: These two rows map to `RequestProperties.RequestExecuteStart` / `RequestExecuteStop`, whose source comments explicitly say *"Start/Stop of execution of request, most time in there is assumed to be user code"*. Calling this a "load phase" is misleading and conflicts with the existing `AdapterLoadStart/Stop` rows (already documented as "Timestamp when test adapter loading started/completed"), which are the actual load phase.

```suggestion
| (main) | Timestamp when test request execution started. |
| (main) | Timestamp when test request execution completed. |
```

| (main) | Boolean for mapping inconclusive tests to failed. |
| (main) | Boolean for mapping non-runnable tests to failed. |
| (main) | Boolean for ordering tests alphabetically by name. |

> | (main) | Boolean indicating if test parallelization is enabled. |

Reviewer: `MSTestTelemetryDataCollector.AddSettingsMetrics` also emits two parallelization items that aren''t documented in this PR — should be added alongside this row if the intent is parity with production telemetry CSV:

- `mstest.setting.parallelization_scope` — string (`MethodLevel` / `ClassLevel`).
- `mstest.setting.parallelization_workers` — number of parallel workers (only emitted when configured).
